### PR TITLE
WELD-2198 replace guava - collection utilities.

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

**Comment by tremes** [comment]
retest please.

File: tests-arquillian/src/test/java/org/jboss/weld/tests/annotatedType/interceptors/SetupExtension.java [hunk 1]
```diff
@@ -55,7 +53,16 @@ public boolean isAnnotationPresent(Class<? extends Annotation> annotationType) {
             @Override
             @SuppressWarnings({ "unchecked", "rawtypes" })
             public Set<AnnotatedMethod<? super Box>> getMethods() {
-                return Sets.newHashSet(Collections2.transform(delegate().getMethods(), new MethodWrappingFunction()));
+                ImmutableSet.Builder builder = ImmutableSet.builder();
```
Comment (mkouba) [inline tests-arquillian/src/test/java/org/jboss/weld/tests/annotatedType/interceptors/SetupExtension.java]: I wouldn't use `ImmutableSet` at all. Also `InterceptedAnnotatedTypeTest.getDeployment()` declares a dependency on guava...

File: impl/src/main/java/org/jboss/weld/util/BeanMethods.java [hunk 2]
```diff
@@ -325,7 +324,13 @@ protected void duplicateMethod(EnhancedAnnotatedMethod<?, ? super T> method) {
      * @see https://issues.jboss.org/browse/WELD-2019
      */
     public static <T> Collection<EnhancedAnnotatedMethod<?, ? super T>> filterMethods(final Collection<EnhancedAnnotatedMethod<?, ? super T>> methods) {
-        return Collections2.filter(methods, METHOD_FILTER_PREDICATE);
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/util/BeanMethods.java]: `METHOD_FILTER_PREDICATE` is then useless, or?

File: impl/src/main/java/org/jboss/weld/util/BeanMethods.java [hunk 3]
```diff
@@ -325,7 +324,13 @@ protected void duplicateMethod(EnhancedAnnotatedMethod<?, ? super T> method) {
...
+        List<EnhancedAnnotatedMethod<?, ? super T>> filteredMethods = new ArrayList<>();
+        for (EnhancedAnnotatedMethod<?, ? super T> method : methods) {
+            if (!method.getJavaMember().isBridge() && !method.getJavaMember().isSynthetic()) {
+                filteredMethods.add(method);
+            }
+        }
+        return ImmutableList.copyOf(filteredMethods);
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/util/BeanMethods.java]: I wonder whether `java.util.Collections.unmodifiableList(List<? extends T>)` wouldn't be more appropriate (and more performant) here. It seems the returned collection is not stored anywhere.

File: impl/src/main/java/org/jboss/weld/util/BeanMethods.java (hunk 2 above)
Comment (tremes) [inline impl/src/main/java/org/jboss/weld/util/BeanMethods.java]: True

File: impl/src/main/java/org/jboss/weld/util/BeanMethods.java (hunk 3 above)
Comment (tremes) [inline impl/src/main/java/org/jboss/weld/util/BeanMethods.java]: Yes could be.

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request closed without merging**